### ISSUE TITLE
New reddit layout and its API (was: Failed to progress errors and incomplete titles)

### ISSUE THREAD

**Issue by johanbluecreek** [johanbluecreek/reddytt#8]
**EDIT**: The third post has some more accurate info of what is going on.

The message

```
Reddytt: Could not identify 'after'-variable to progress deeper.
```

is sometimes displayed, after the fetching has taken quite some time, so is there a timeout that happens? In any case, it is fine since the videos starts playing but the titles are wrong after this. The are always of the type

```
youtu.be/UY9P0Q...
```

literally, it is broken off by ellipsis and not the whole ID is displayed. Very annoying. This has been observed at least three times.

Issue would be considered resolved when the titles are displayed correctly. The `progress deeper` message is fine, just fix the titles.

**Comment by johanbluecreek** [comment]
There is currently a workaround in master. It checks if `after` is among any of the links, of not, it retries fetching the Reddit-page.

This should probably be improved for several reasons:

* Reddit is changing their profiles, so this may be due to some new layout they are pushing in combination with this.
* This only checks the `after` variable when it is the link-titles we want fixed. Now, the page will be fetched a bunch every time an `after` does not exist, which might be because there isn't another page.

So I'm letting the issue remain open.

**Comment by johanbluecreek** [comment]
Through some experiments I did find that it is the new reddit layout of the homepage that screws up the way titles are found. Since I cannot expect the old layout to always be the fallback (as reddytt does now thanks to 0016e13dd5741c80d6015ba299f8f1680c5dfe8f), it would be best to use the API. The difficulties I have with implementing the API calls are described in the ffd8a47a7abe3da4b00b61ce28c1680e7d8d80e7 commit. Because of the annoying OAuth2, it might as well be easier to use `bs4` again, but since I have found no way of matching title with link yet...

Help would be appreciated. 

> No one hears you.
